### PR TITLE
API docs for RestoreTeam

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary

This PR provides docs for new restore team endpoint.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-23724

### CODE REVIEW

File: v4/source/teams.yaml [hunk 1]
```diff
@@ -429,6 +429,46 @@
...
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "404":
+          $ref: "#/components/responses/NotFound"
+      x-code-samples:
+        - lang: Go
+          source: |
+            import "github.com/mattermost/mattermost-server/model"
```
Comment (saturninoabril) [inline v4/source/teams.yaml:467]: Non-blocking and can be on separate PR if it's true.  Should this (and all other instances) to be updated using `.../v5/model`?

Comment (wiggin77) [reply v4/source/teams.yaml:467]: Yes, I agree.  We'll have to create a new ticket for that and update all of them.

Comment (wiggin77) [reply v4/source/teams.yaml:467]: Hi @saturninoabril , I added a ticket to platform backlog to address this.
